torso-bundle.js

```diff
@@ -1431,32 +1509,44 @@
      * Rebuilds the html for this view's element. Should be able to be called at any time.
      * Defaults to using this.templateRender. Assumes that this.template is a javascript
      * function that accepted a single JSON context.
+     * The render method returns a promise that resolves when rendering is complete. Typically render
+     * is synchronous and the rendering is complete upon completion of the method. However, when utilizing
+     * transitions/animations, the render process can be asynchronous and the promise is useful to know when it has finished.
      * @method render
+     * @return {Promise} a promise that when resolved signifies that the rendering process is complete.
      */
     render: function() {
-      this.unplug();
-      if (this.template) {
-        this.templateRender(this.$el, this.template, this.prepare());
-      }
-      this.plug();
+      var renderPromises = [];
+      this.trigger('render-begin');
+      addPromises(renderPromises, this.prerender());
+      this.trigger('render-before-dom-replacement');
+      this.__createDOM();
       this.delegateEvents();
+      this.trigger('render-after-dom-replacement');
+      addPromises(renderPromises, this.attachTrackedViews());
+      this.__injectionSiteMap = {};
```

Comment:
we clear the injection site map here?  That seems weird to me to clear it after attaching all the views and swapping out the injection sites.  I was expecting it to be a map that persists between rendering so you can revert the injection sites.